src/ImmersedBoundaries/immersed_grid_metrics.jl

```diff
@@ -34,6 +36,11 @@ end
 @inline Δzᵃᵃᶠ(i, j, k, ibg::IBG) = Δzᵃᵃᶠ(i, j, k, ibg.underlying_grid)
 
 return_metrics(grid::IBG) = return_metrics(grid.underlying_grid)
-xspacings(X, grid::IBG) = xspacings(X, grid.underlying_grid)
-yspacings(Y, grid::IBG) = yspacings(Y, grid.underlying_grid)
-zspacings(Z, grid::IBG) = zspacings(Z, grid.underlying_grid)
+
+xspacing(grid::IBG, args...) = xspacing(grid.underlying_grid, args...)
+yspacing(grid::IBG, args...) = yspacing(grid.underlying_grid, args...)
+zspacing(grid::IBG, args...) = zspacing(grid.underlying_grid, args...)
```

Comment:
these won't work I believe...

```suggestion
xspacing(i, j, k, grid::IBG, args...) = xspacing(i, j, k, grid.underlying_grid, args...)
yspacing(i, j, k, grid::IBG, args...) = yspacing(i, j, k, grid.underlying_grid, args...)
zspacing(i, j, k, grid::IBG, args...) = zspacing(i, j, k, grid.underlying_grid, args...)
```